### DISCUSSION TITLE
Fix for "Set the Ultimate Performance power plan as active" not working

### DISCUSSION THREAD

**Discussion by Alvaro-TMPC** [memstechtips/UnattendedWinstall#101] [Show and tell]
Hi!

I've been testing the standard autounattend file, and everything works perfectly except for the activation of the Ultimate Performance Power Plan:

Enable the Ultimate Performance power plan
powercfg -duplicatescheme e9a42b02-d5df-448d-aa00-03f14749eb61

Set the Ultimate Performance power plan as active
powercfg -setactive e9a42b02-d5df-448d-aa00-03f14749eb61

This code creates the Ultimate Performance Power Plan but does not activate it properly. The issue is that when the scheme is duplicated, a new GUID is assigned, so the -setactive command does not work as expected.

![image](https://github.com/user-attachments/assets/964b031c-67d8-4feb-b91b-7121c299903f)

I've devised a solution to resolve this issue with the following code:

$maxPerformanceScheme = powercfg -duplicatescheme e9a42b02-d5df-448d-aa00-03f14749eb61
$guid = [regex]::Match($maxPerformanceScheme, '[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}').Value
powercfg -setactive $guid

I have tested it locally and it works correctly. I use Windows in Spanish, but I understand that in any language it should work.

**Comment by memstechtips** [comment]
Thanks so much for this. 

I will test it out myself and see if it works. 
If it does, I will incorporate it.

**Comment by XerXes777** [comment]
Dont work on laptops with a new intel cpu and modern standby. [Info](https://h30434.www3.hp.com/t5/Business-Notebooks/HP-Zbook-G10-sleep-and-modern-standby-issues/m-p/8898129/highlight/true#M171275)

> Unfortunately Intel (the CPU manufacturer) has removed the ability to go to S3 sleep on all its processors, starting with 11th gen Tiger Lake CPUs. This is because it believes Modern Standby is......well the modern way for a system to Sleep. Even on 11/12/13th gen laptops that don't have S3 removed from the BIOS, if you disable Modern Standby and try S3 sleep, it will seem to work fine, but will never resume from standby. This is why HP has (reportedly) started removing S3 on a BIOS level as well. Therefore, I'm afraid it's not possible to go to good old-fashioned Sleep on your system. What all of us do nowadays (this includes myself as well) is use hibernation.

 

> I've only seen one thread on the HP forum where someone implied that hibernation (S4) is completely disabled on the system (https://h30434.www3.hp.com/t5/Business-Notebooks/Modern-sleep-problems-in-Linux/td-p/8884504). I haven't been able to confirm if the recent Elitebooks and ZBooks (G9/G10) have removed this from their BIOS along with S3. It seems more probable that the cause of the problem with hibernation is Modern Standby. This is because Modern Standby doesn't allow S4 to work correctly. Microsoft explains it [here](https://learn.microsoft.com/en-us/windows-hardware/design/device-experiences/display--sleep--and-hibernate-idle-timers): "Although Modern Standby systems support Hibernate (S4) state, it is not entered automatically after a fixed amount of time in sleep. Instead, Windows manages Hibernate intelligently, only using it when required to preserve user's battery life. That's because modern standby is a consistently low-power state with long battery life." Therefore, what you need to do is completely disable Modern Standby, and simply keep Hibernate. Bear in mind that after you disable it, you must change all your Windows settings to "never sleep".

 

> Disabling Modern Standby does not involve an option in either BIOS or Windows settings. It is hidden, and you have to change a value in your Windows registry. Many guides exist online, here's one: [How to Disable Modern Standby (Win10/11)](https://www.makeuseof.com/windows-disable-modern-standby/) . Use the powercfg /a command mentioned on the page to verify that S4 (hibernate) is still an option on your system.

 

> If you can't or don't want to DISABLE Modern Standby, Microsoft has provided a complicated way to tweak the S4 timers. I haven't tried it myself, but the information is on this page: [Adaptive hibernate overview](https://learn.microsoft.com/en-us/windows-hardware/customize/power-settings/adaptive-hibernate) .

**Comment by often-called-pk** [comment]
i tried the following command in terminal with admin privilege - 
$maxPerformanceScheme = powercfg -duplicatescheme e9a42b02-d5df-448d-aa00-03f14749eb61
$guid = [regex]::Match($maxPerformanceScheme, '[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}').Value
powercfg -setactive $guid

but every time i restart the ultimate power goes away.
how do i make it permanent???
